### PR TITLE
Proto changes for client library generation

### PR DESCRIPTION
**Pull request by Katee**
* Remove sub packages under `grafeas.v1` (e.g. `grafeas.v1.vulnerability`, `grafeas.v1.attestation`)
* Use `grafeas_proto_proto` as `go_package` everywhere
* Rename `Details` fields to `<KIND>Occurrence`
* Use `<KIND>Note` for note details and flatten for some kinds
* Put `source.proto` in `provenance.proto` (Java client generation encountered an error "io/grafeas/v1/Source.java: Tried to write the same file twice" without this change)
* Remove unused timestamp import from `discovery.proto`
* Add make target for the v1 protos

### CODE REVIEW

File: proto/v1/grafeas.proto [hunk 1]
```diff
@@ -208,20 +208,20 @@ message Occurrence {
   // resource.
   oneof details {
     // Describes a security vulnerability.
-    grafeas.v1.vulnerability.Details vulnerability = 8;
+    grafeas.v1.VulnerabilityOccurrence vulnerability = 8;
```
Comment (vtsao) [inline proto/v1/grafeas.proto]: Do you think we should we rename all these fields to be [KIND]Occurrence as well?

Comment (aysylu) [reply proto/v1/grafeas.proto]: I'd vote for now, for brevity. However, here are suggestions for renaming, to be consistent with the overall naming strategy:
* `derived_image` -> `image`
* `installation` -> `package`
* `discovered` -> `discovery`

File: proto/v1/grafeas.proto [hunk 2]
```diff
@@ -279,19 +279,19 @@ message Note {
...
   oneof type {
     // A note describing a package vulnerability.
-    grafeas.v1.vulnerability.Vulnerability vulnerability = 10;
+    grafeas.v1.VulnerabilityNote vulnerability = 10;
     // A note describing build provenance for a verifiable build.
-    grafeas.v1.build.Build build = 11;
+    grafeas.v1.BuildNote build = 11;
     // A note describing a base image.
-    grafeas.v1.image.Basis base_image = 12;
+    grafeas.v1.ImageNote base_image = 12;
```
Comment (aysylu) [inline proto/v1/grafeas.proto]: Ditto here for naming strategy of the fields.

File: proto/v1/grafeas.proto (hunk 1 above)
Comment (Katee) [reply proto/v1/grafeas.proto]: @vtsao do you mean changing all lines to (for example) `grafeas.v1.VulnerabilityOccurrence vulnerability_occurrence = 8;`?

Comment (aysylu) [reply proto/v1/grafeas.proto]: Oops, typo, should have been `I'd vote for no`.

**Comment by Katee** [comment]
I force pushed a change to remove the proto re-generation from this PR.

File: proto/v1/grafeas.proto (hunk 1 above)
Comment (vtsao) [reply proto/v1/grafeas.proto]: Yeah this LGTM.

**Comment by aysylu** [comment]
LGTM, thanks for implementing the requested changes.

**Pull request merged**